File: Spring-boot-backend/package-master-service/src/main/java/com/packagemaster/service/model/QuoteRequest.java
package com.packagemaster.service.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;

@Entity
@Table(name = "quote_requests", indexes = {
    @Index(name = "ix_quote_requests_email", columnList = "email"),
    @Index(name = "ix_quote_requests_status", columnList = "status"),
    @Index(name = "ix_quote_requests_product_type", columnList = "product_type"),
    @Index(name = "ix_quote_requests_created_at", columnList = "created_at")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuoteRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(name = "company_name", length = 200)
    private String companyName;

    @Column(name = "phone_number", nullable = false, length = 20)
    private String phoneNumber;

    @Column(nullable = false, length = 200)
    private String email;

    @Column(name = "product_type", nullable = false, length = 100)
    private String productType;

    @Column(nullable = false, precision = 18, scale = 2)
    private BigDecimal length;

    @Column(nullable = false, precision = 18, scale = 2)
    private BigDecimal width;

    @Column(nullable = false, precision = 18, scale = 2)
    private BigDecimal height;

    @Column(nullable = false)
    private Integer quantity;

    @Column(length = 100)
    private String color;

    @Column(name = "printing_required", nullable = false)
    private boolean printingRequired;

    @Column(name = "delivery_location", nullable = false, length = 300)
    private String deliveryLocation;

    @Column(name = "expected_timeline", nullable = false, length = 200)
    private String expectedTimeline;

    @Column(name = "additional_notes", columnDefinition = "text")
    private String additionalNotes;

    @Column(name = "reference_image_path", length = 500)
    private String referenceImagePath;

    @Column(name = "artwork_path", length = 500)
    private String artworkPath;

    @Builder.Default
    @Column(nullable = false, length = 50)
    private String status = "NEW";

    @Column(columnDefinition = "text")
    private String notes;

    @Builder.Default
    @Column(name = "created_at", nullable = false)
    private Instant createdAt = Instant.now();
}

File: Spring-boot-backend/PackagingPortal/PackagingPortal.Api/Models/QuoteRequest.cs
using System.ComponentModel.DataAnnotations;

namespace PackagingPortal.Api.Models;

public class QuoteRequest
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(200)]
    public string? CompanyName { get; set; }

    [Required]
    [MaxLength(20)]
    public string PhoneNumber { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string ProductType { get; set; } = string.Empty;

    [Required]
    public decimal Length { get; set; }

    [Required]
    public decimal Width { get; set; }

    [Required]
    public decimal Height { get; set; }

    [Required]
    public int Quantity { get; set; }

    [MaxLength(100)]
    public string? Color { get; set; }

    public bool PrintingRequired { get; set; }

    [Required]
    [MaxLength(300)]
    public string DeliveryLocation { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    public string ExpectedTimeline { get; set; } = string.Empty;

    public string? AdditionalNotes { get; set; }

    [MaxLength(500)]
    public string? ReferenceImagePath { get; set; }

    [MaxLength(500)]
    public string? ArtworkPath { get; set; }

    [MaxLength(50)]
    public string Status { get; set; } = "NEW";

    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}